creator-node/src/middlewares.js

```diff
@@ -32,6 +36,45 @@ async function authMiddleware (req, res, next) {
     wallet: cnodeUser.walletPublicKey,
     cnodeUserUUID: cnodeUserUUID
   }
+
+  next()
+}
+
+/** Conditional logic if call is coming from another creator node */
+async function crossCnodeAuth (req, res, next) {
+  if (!req.session || !req.session.cnodeUser) {
+    return sendResponse(req, res, errorResponseBadRequest('req.session.cnodeUser not provided'))
+  }
+
+  if (req.session.cnodeUser.isCreatorNode) {
+    // CNode must specify which artist it is making call for
+    const artistWallet = req.query.artistWallet
+    if (!artistWallet) {
+      return sendResponse(req, res, errorResponseBadRequest('ArtistWallet not provided'))
+    }
+
+    // Fetch own service endpoint from local config or eth contract
+    let serviceEndpoint
+    try {
+      serviceEndpoint = await _getOwnEndpoint(req)
+    } catch (e) {
+      return sendResponse(req, res, errorResponseServerError(e))
+    }
+
+    // Fetch artist replica set from chain
+    let artistReplicaSet
+    try {
+      artistReplicaSet = await _getCreatorNodeEndpoints(req, artistWallet)
```

Comment:
i know this functionality doesn't currently exist so i know why you did it this way, but i think in the future this should be spID

Reply:
per discussion, going to standardize all to spID